src/GenerativeDesigns/distancebased.jl

```diff
@@ -50,20 +52,97 @@ end
 """
     Entropy(labels; prior)
 
-Return a function of `weights` that computes the percentage of information entropy, compared to the entropy calculated with respect to a specified `prior`.
+Return a function of `weights` that computes the fraction of information entropy, relative to the entropy calculated with respect to a specified `prior`.
 """
 function Entropy(labels; prior)
     @assert elscitype(labels) <: Multiclass "labels must be of `Multiclass` scitype, but `elscitype(labels)=$(elscitype(labels))`"
     initial = compute_entropy(labels; weights = prior)
     return (weights -> compute_entropy(labels; weights) / initial)
 end
 
+# Return a function that calculates the sum of distances in each row, column-wise, and applies weights based on the prior.
+function sum_of_distances(data::DataFrame, targets::Vector, distances; prior::Weights)
+    function (evidence::Evidence)
+        if isempty(evidence)
+            return zeros(nrow(data))
+        else
+            array_distances = zeros((nrow(data), length(evidence)))
+            for (i, colname) in enumerate(keys(evidence))
+                if colname ∈ targets
+                    continue
+                else
+                    array_distances[:, i] .=
+                        distances[colname](evidence[colname], data[!, colname]; prior)
+                end
+            end
+
+            distances_sum = vec(sum(array_distances; init = 0.0, dims = 2))
+
+            return distances_sum
+        end
+    end
+end
+
 """
-    DistanceBased(data, target, uncertainty, similarity=Exponential(), distances=Dict(); prior=ones(nrow(data)))
+    MahalanobisDistance(; diagonal=0)
+
+Returns a function that computes [Mahalanobis distance](https://en.wikipedia.org/wiki/Mahalanobis_distance) between each row of `data` and the evidence.
+For a singular covariance matrix, consider adding entries to the matrix's diagonal via the `diagonal` keyword.
+
+# Arguments
+
+  - `diagonal`: A scalar or vector to be added to the diagonal entries of the covariance matrix.
+
+# Returns
+
+It returns a high-level function of `(data, targets, prior)`.
+When called, that function will return an internal function `compute_distances` that takes an `Evidence` and computes the Mahalanobis distance based on the input data and the evidence.
+"""
+function MahalanobisDistance(; diagonal = 0)
```

Comment:
```julia
using Distances
using LinearAlgebra
using StatsBase

function mahalanobisdistance(; diagonal=0)
    function (data, targets, prior)
        non_targets = setdiff(names(data), targets)
        # Use DataFrame accessors for columns and convert them to Matrix if needed.
        data_matrix = convert(Matrix, data[:, non_targets])
        # Calculate the weighted covariance matrix and adjust the diagonal if specified.
        σ = cov(data_matrix, Weights(prior))
        if diagonal != 0
            if diagonal isa Number
                σ += I * diagonal
            else
                @assert length(diagonal) == size(σ, 1) "Length of diagonal must match the dimensions of the covariance matrix."
                diag_indices = diagind(σ)
                σ[diag_indices] += diagonal
            end
        end
        
        # Check if covariance matrix is singular.
        if rank(σ) < length(non_targets)
            error("Singular covariance matrix σ (rank $(rank(σ))). Consider adding entries to diagonal via `diagonal` keyword.")
        end

        # Pre-compute the inverse of the covariance matrix for performance.
        inv_σ = inv(σ)
        
        compute_distances = function (evidence)
            if isempty(evidence)
                return zeros(size(data, 1))
            else
                # Construct a vector for the evidence, with zeros for missing non-target columns.
                vec_evidence = [get(evidence, col, 0) for col in non_targets]
                # Calculate the Mahalanobis distance for each row in the data.
                distances = [mahalanobis(vec_row, vec_evidence, inv_σ) for vec_row in eachrow(data[:, non_targets])]
                return distances
            end
        end
        return compute_distances
    end
end
```
`Distance.jl` has the internal function for the calculation. I refactor it for the row_wise Mahalanobis distance calculation

Reply:
Hmm, I am not sure if it worth to introduce additional dependency, especially considering that the calculation involved by Distances.jl is just `dot(z, Q * z)`.

Reply:
Yeah, it is just a matrix multiplication and a dot product. Updated to use the same code as Distances.jl does.